Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -155,34 +152,20 @@ void FindSXPeaks::exec() {
     double rightBinEdge = *std::next(leftBinPosition);
     double tof = 0.5 * (leftBinEdge + rightBinEdge);
 
-    Geometry::IDetector_const_sptr det;
-    try {
-      det = localworkspace->getDetector(static_cast<size_t>(i));
-    } catch (Mantid::Kernel::Exception::NotFoundError &) {
-      // Catch if no detector. Next line tests whether this happened - test
-      // placed
-      // outside here because Mac Intel compiler doesn't like 'continue' in a
-      // catch
-      // in an openmp block.
-    }
     // If no detector found, skip onto the next spectrum
-    if (!det)
+    if (!spectrumInfo.hasDetectors(static_cast<size_t>(i))) {
       continue;
+    }
+    const auto &det = spectrumInfo.detector(static_cast<size_t>(i));
 
-    double phi = det->getPhi();
+    double phi = det.getPhi();
     if (phi < 0) {
       phi += 2.0 * M_PI;
     }
 
-    double th2 = det->getTwoTheta(sample, L1);
-
     std::vector<int> specs(1, i);
 
-    Mantid::Kernel::V3D L2 = det->getPos();
-    L2 -= sample;
-
-    SXPeak peak(tof, th2, phi, *maxY, specs, l1 + L2.norm(), det->getID(),
-                localworkspace->getInstrument());
```

Comment:
I'm not most expert on the peaks side of things so some one else should comment for that but in the case of a group of detectors it does not make any sense to try and pull out an intensity from an individual detector - that information has been lost when the grouping is performed (whether hardware or software). 

@soininen Do the tests actually use any grouping anywhere?